ansible/roles/test/files/ptftests/vlan_test.py

```diff
@@ -0,0 +1,268 @@
+        vlan_ports_list = self.vlan_ports_list
+        d = defaultdict(list)
+        for vlan_port in vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) == vlan_port["pvid"]:
+                    iface = "eth%d" % vlan_port["port_index"]
+                else:
+                    iface = "eth%d.%s" % (vlan_port["port_index"], permit_vlanid)
+                d[iface].append(vlan_port["permit_vlanid"][str(permit_vlanid)]["peer_ip"])
+        with open('/tmp/from_t1.json', 'w') as file:
+            json.dump(d, file)
+
+    #--------------------------------------------------------------------------
+    def tearDown(self):
+        logging.info("VLAN test ending ...")
+
+        self.log("Delete VLAN intf")
+        for vlan_port in self.vlan_ports_list:
+            for permit_vlanid in vlan_port["permit_vlanid"].keys():
+                if int(permit_vlanid) != vlan_port["pvid"]:
+                    self.shell(["ip", "link", "delete", "eth%d.%d"%(vlan_port["port_index"], int(permit_vlanid))])
+
+        self.log("Stop arp_responder")
+        self.shell(["supervisorctl", "stop", "arp_responder"])
+
+        pass
+
+    #--------------------------------------------------------------------------
+    def build_icmp_packet(self, vlan_id, 
+                          src_mac="00:22:00:00:00:02", dst_mac="00:22:00:00:00:01", 
+                          src_ip="192.168.0.1", dst_ip="192.168.0.2", ttl=64):
+        pkt = simple_icmp_packet(pktlen=100 if vlan_id == 0 else 104,
+                                 eth_dst=dst_mac,
+                                 eth_src=src_mac,
+                                 dl_vlan_enable=False if vlan_id == 0 else True,
+                                 vlan_vid=vlan_id,
+                                 vlan_pcp=0,
+                                 ip_src=src_ip,
+                                 ip_dst=dst_ip,
+                                 ip_ttl=ttl)
+        return pkt
+
+
+    #--------------------------------------------------------------------------
+    def send_icmp_packet(self, vlan_port, vlan_id=0,
+                       src_mac="00:22:00:00:00:02", dst_mac="00:22:00:00:00:01", 
+                       src_ip="192.168.0.1", dst_ip="192.168.0.2", ttl=64):
+        pkt = self.build_icmp_packet(vlan_id, src_mac, dst_mac, src_ip, dst_ip, ttl)
+        self.log("Send " + ("untagged" if vlan_id == 0 else "tagged(%d)"%vlan_id) 
+                 + " packet from " + str(vlan_port["port_index"]) + "...")
+        self.log("%s -> %s, %s -> %s"%(src_mac, dst_mac, src_ip, dst_ip))
+        send(self, vlan_port["port_index"], pkt)
+
+    #--------------------------------------------------------------------------
+    def verify_icmp_packets(self, vlan_port, vlan_id,
+                          src_mac="00:22:00:00:00:02", dst_mac="00:22:00:00:00:01", 
+                          src_ip="192.168.0.1", dst_ip="192.168.0.2", ttl=64):
```

Comment:
you already have default paramaters in build_icmp_packet, can we avoid duplication here?